service/src/main/java/bio/terra/workspace/db/ApplicationDao.java

```diff
@@ -76,6 +107,176 @@ public void createApplication(WsmApplication app) {
     jdbcTemplate.update(sql, params);
   }
 
+  /**
+   * Disable an application in a workspace. It is not an error to disable an already disabled
+   * application.
+   *
+   * @param workspaceId workspace of interest
+   * @param applicationId application of interest
+   * @return workspace-application object
+   */
+  @WriteTransaction
+  public WsmWorkspaceApplication disableWorkspaceApplication(UUID workspaceId, UUID applicationId) {
+
+    // Validate that the application exists
+    getApplication(applicationId);
+
+    final String sql =
+        "DELETE FROM enabled_application"
+            + " WHERE workspace_id = workspace_id AND application_id = :application_id";
```

Comment:
missing`:` in front of `workspace_id` here! I think this clause is always true otherwise

Reply:
Hah! And the tests all worked anyway :)